icaruscode/Timing/PMTBeamSignalsExtractor_module.cc

```diff
@@ -0,0 +1,541 @@
+////////////////////////////////////////////////////////////////////////
+// Class:       PMTBeamSignalsExtractor
+// Plugin Type: producer (Unknown Unknown)
+// File:        PMTBeamSignalsExtractor_module.cc
+//
+// Generated at Sun Feb 11 11:37:14 2024 by Matteo Vicenzi using cetskelgen
+// from  version .
+////////////////////////////////////////////////////////////////////////
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Principal/Run.h"
+#include "art/Framework/Principal/SubRun.h"
+#include "canvas/Utilities/InputTag.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "fhiclcpp/types/Atom.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+#include "art_root_io/TFileService.h"
+
+#include "sbnobj/Common/Trigger/ExtraTriggerInfo.h"
+#include "icaruscode/Decode/ChannelMapping/IICARUSChannelMap.h"
+#include "icaruscode/IcarusObj/PMTWaveformTimeCorrection.h"
+#include "icaruscode/IcarusObj/PMTBeamSignal.h"
+#include "lardata/DetectorInfoServices/DetectorClocksService.h"
+#include "lardataalg/DetectorInfo/DetectorTimings.h"
```

Comment:
If you accept the suggestion to read from the data product rather than from the service, you'll have to trade:
```suggestion
#include "lardataobj/RawData/TriggerData"
```

Reply:
Done!